clients/drcachesim/tools/view.cpp

```diff
@@ -103,6 +103,34 @@ view_t::initialize()
 bool
 view_t::process_memref(const memref_t &memref)
 {
+    // Even for -skip_refs we need to process the up-front version and type.
+    if (memref.marker.type == TRACE_TYPE_MARKER) {
+        switch (memref.marker.marker_type) {
+        case TRACE_MARKER_TYPE_VERSION:
+            // We delay printing until we know the tid.
+            if (trace_version_ == -1)
+                trace_version_ = static_cast<int>(memref.marker.marker_value);
+            else if (trace_version_ != static_cast<int>(memref.marker.marker_value)) {
+                error_string_ = std::string("Version mismatch across files");
+                return false;
+            }
+            break;
+        case TRACE_MARKER_TYPE_FILETYPE:
+            // We delay printing until we know the tid.
+            prev_filetype_ = memref.marker.marker_value;
```

Comment:
Why don't we have a check like above that fails if a different file type is found later? I thought we can have only one file type in a trace.

Reply:
You mean like the version check?  Done.

Reply:
Yes, thanks.